### ISSUE TITLE
Problem: join() method limitations

### ISSUE THREAD

**Issue by newliteral** [zeromq/libzmq#3035]
# Issue description
I'm using newest radio/dish sockets. The join() method accepts group name until 15 characters. Will the restriction be saved in the future? Thanks.

# Environment

* libzmq version (commit hash if unreleased): 4.2.3 with drafts
* OS: Lubuntu 17.10

# Minimal test code / Steps to reproduce the issue

```python
import zmq

ctx = zmq.Context.instance()
dish = ctx.socket(zmq.DISH)

dish.bind('udp://239.0.0.1:8008')
dish.join('0123456789abcde')  # Ok
dish.join('0123456789abcdef') # zmq.error.ZMQError: Invalid argument
```

**Comment by bluca** [comment]
Don't think so as it's part of the protocol - in what use case would more be needed?

**Comment by newliteral** [comment]
I'm using zmq for events multicasting on the HTML5 SSE connections. For my opinion, minimal group length must be about 24 bytes for storing UUID with small prefix. Increasing valid length is a technical problem?

**Comment by bluca** [comment]
The way this was implemented is such that the group is part of the message metadata, rather than being a different frame. That's because with RADIO/DISH there is (intentionally) no support for metadata.

This limits the size to 16 bytes, as that's how much can be crammed in the available space: https://github.com/zeromq/libzmq/blob/master/src/msg.hpp#L180

This is very unlikely to change for that reason, so if you need more than 16 bytes use PUB/SUB rather than RADIO/DISH.

**Comment by newliteral** [comment]
Well. Please, document this.
Standard PUB/SUB socket has no required functions for me. Thanks.

**Comment by bluca** [comment]
Yes it should definitely be documented - most of the DRAFT APIs are missing documentation, as they are still under development. I opened https://github.com/zeromq/libzmq/issues/3038 to track this.

> Standard PUB/SUB socket has no required functions for me. Thanks.

Not sure what you mean by that? What's missing?

**Comment by newliteral** [comment]
I need multicasting in "many-to-many" mode, where each server process is a both consumer and a producer. Losses are acceptable.
In case TCP PUB/SUB, it's possible with [proxy](http://zguide.zeromq.org/page%3aall#The-Dynamic-Discovery-Problem) or bus topology, but it is complex, thread-unsafe and potencially blocking if the consumer's network buffer is overflowed. And for performance reasons it is a bit of a lost cause.
Last question: what guarantees a RADIO/DISH (ordering, duplicate protection etc)?

**Comment by bluca** [comment]
> Last question: what guarantees a RADIO/DISH (ordering, duplicate protection etc)?

Not sure sorry, this is a question for @somdoron
